lib/wallaby/browser.ex

```diff
@@ -857,6 +857,17 @@ defmodule Wallaby.Browser do
     end
   end
 
+  defp do_get(query, elements) do
+    case {Query.get_number(query), length(elements)} do
+      {nil,_} ->
+        {:ok, elements}
+      {n, count} when n > 0 and n <= count ->
```

Comment:
I did put a bit of thought into this, so let me try to explain my reasoning...

(But before I begin, perhaps we can let your inner programmer sleep easy at nights knowing that many programming languages use 1-based indexing. I have programmed in a few of them myself - including, interestingly enough, Erlang, which is mainly what I have been using for the last 6 years of so.

Maybe that is what is influencing me, but I think it goes rather deeper than that. Another example is css, which was the inspiration for your `nth` suggestion. In css, `nth` is 1-based, and I think this reflects a much wider usage. The use of 0-based is in truth not an _index_, but an _offset_. The beauty of offsets in a programming language is that you can add them and you will get the expected result. But what we are dealing with here is not a programming language but a DSL, and the domain under consideration is a web page. Picking the 3rd item in a list by specifying that you want item at position 2 I would think is counter-intuitive to most people. Instances of 0-based addressing outside of programming languages are very rare indeed. I can not think of a single one, off the top of my head. Certainly there is nobody living at number 0 on my street!

There may be all kinds of technical people who are not programmers - web designers, perhaps, or maybe full-time testers - who end up using Wallaby. I certainly hope so, and they are the ones I am thinking about. It is to make this technology accessible to such people is one of the main reasons that we have DSLs in the first place.

Programmers will deal with it whichever we choose - especially if they have experience of both systems - but for 'normal' people 1-based is the natural way.

So have I done a good enough selling job? I hope so. But, as always, this is your gig, and the final call is yours.